projects/ADuCM360_demo_cn0411/include/AD5683.h

```diff
@@ -0,0 +1,111 @@
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef _AD5683_H_
+#define _AD5683_H_
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define AD5683_CMD_NOP					0x0     // No operation.
+#define AD5683_CMD_WR_IN_REG			0x1 	// Write Input Register.
+#define AD5683_CMD_WR_DAC_REG           0x2 	// Update DAC Register.
+#define AD5683_CMD_WR_DAC_IN_REG     	0x3		// Write DAC and Input Register.
+#define AD5683_CMD_WR_CTRL_REG			0x4 	// Write Control Register.
+#define AD5683_CMD_RB_IN_REG			0x5		// Readback input register.
+
+/*************************** Write Control Register Bits **********************/
+
+#define AD5683_DCEN(x)					(((((x) & 0x1) << 0) << 10) & 0xFC00)
+#define AD5683_CFG_GAIN(x)				(((((x) & 0x1) << 1) << 10) & 0xFC00)
+#define AD5683_REF_EN(x)				(((((x) & 0x1) << 2) << 10) & 0xFC00)
+#define AD5683_OP_MOD(x)				(((((x) & 0x3) << 3) << 10) & 0xFC00)
+#define AD5683_SW_RESET(x)				(((((x) & 0x1) << 5) << 10) & 0xFC00)
+
+/******************************************************************************/
+/**************************** Variable Declarations ***************************/
+/******************************************************************************/
+typedef struct {
```

Comment:
typedef's are no longer accepted. 
Leave it as 
struct ad5683_device {
}
The same for enums.